include/mppic/optimizer.hpp

```diff
@@ -88,62 +91,60 @@ class Optimizer
    * @param twist current robot speed
    * @param state[out] fill state with velocities on each step
    */
-  void updateStateVelocities(
-    models::State & state,
-    const geometry_msgs::msg::Twist & robot_speed) const;
+  void updateStateVelocities(models::State & state) const;
 
-  void
-  updateInitialStateVelocities(
-    models::State & state,
-    const geometry_msgs::msg::Twist & robot_speed) const;
+  void updateInitialStateVelocities(models::State & state) const;
 
   /**
    * @brief predict velocities in state_ using model
    * for time horizont equal to time_steps_
    */
   void propagateStateVelocitiesFromInitials(models::State & state) const;
 
-  xt::xtensor<double, 3> integrateStateVelocities(
-    const models::State & state, const geometry_msgs::msg::PoseStamped & robot_pose) const;
+  xt::xtensor<double, 3> integrateStateVelocities(const models::State & state) const;
 
   /**
    * @brief Update control_sequence_ with state controls weighted by costs
    * using softmax function
-   *
-   * @param trajectories costs, tensor of shape [ batch_size ]
    */
-  void updateControlSequence(const xt::xtensor<double, 1> & costs);
+  void updateControlSequence();
 
-  /**
-   * @brief Get offseted control from control_sequence_
-   *
-   */
-  auto getControlFromSequence(const unsigned int offset);
   geometry_msgs::msg::TwistStamped
-  getControlFromSequenceAsTwist(
-    const unsigned int offset,
-    const builtin_interfaces::msg::Time & stamp);
+  getControlFromSequenceAsTwist(const builtin_interfaces::msg::Time & stamp);
 
   bool isHolonomic() const;
 
-  void setOffset();
+  void setOffset(double controller_frequency);
 
+  void fallback(bool fail);
+
+protected:
   rclcpp_lifecycle::LifecycleNode::WeakPtr parent_;
   std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros_;
   nav2_costmap_2d::Costmap2D * costmap_;
+  std::string name_;
+
   ParametersHandler * parameters_handler_;
 
-  std::string name_;
-  double controller_frequency_{0};
+  std::unique_ptr<MotionModel> motion_model_;
+  CriticManager critic_manager_;
 
   models::OptimizerSettings settings_;
+
   models::State state_;
   models::ControlSequence control_sequence_;
 
-  std::unique_ptr<MotionModel> motion_model_;
-  CriticManager critic_manager_;
-
   xt::xtensor<double, 3> generated_trajectories_;
+  xt::xtensor<double, 2> plan_;
+  xt::xtensor<double, 1> costs_;
+
+  xt::xtensor<double, 3> vx_noises_;
+  xt::xtensor<double, 3> vy_noises_;
+  xt::xtensor<double, 3> wz_noises_;
+
+  models::CriticFunctionData critics_data_ =
```

Comment:
I don't initializing this is meaningful

Reply:
What do you mean? Naming/doc absence?

Reply:
Never mind, that was before I realized these were all references